### DISCUSSION TITLE
What's up with kitty and SSH?

### DISCUSSION THREAD

**Discussion by ldelossa** [kovidgoyal/kitty#4223] [General]
Hey there. 

Was looking for an alternative to Tilix and a terminal I can make theme switch during running.

Got kitty all setup and was digging it, but then realized ssh is... complicated? 

I work on prod servers all day, copying files to and from them is not acceptable. 

Also, I've never seen this requirement from any other terminal I've used before. Why is this the case?

**Comment by page-down** [comment]
terminfo is used to describe the capabilities of the terminal emulator. Command line programs need to know the correct functional instructions in order to be invoked correctly and present the user with the desired result.

Other terminal emulators actually require the correct terminfo as well, except that many use the definition of xterm. There has been almost no development in this area due to little change in functionality, and because of its age, it has long been built into every major system or distribution.

And kitty wants to drive the ongoing development of this area of terminal emulator and text command line interaction. If you need to fix old problems, get rid of historical baggage, and push out new things, there are bound to be incompatibilities. So, the correct terminfo needs to be installed.

Installing terminfo only needs to be done on version updates, and kitty will try to keep it backward compatible. So you won't need to copy the files every time you connect.

If even terminfo is absolutely forbidden on the production server, then maybe kitty is not for you. It is also not recommended to use environment variable definitions other than TERM = xterm-kitty. This is to avoid unforeseen operational errors that can affect the production environment.

**Comment by ldelossa** [comment]
Ah I see. 

Yes, sometimes we need to ssh directly into customer's machines. We aren't allowed to modify these in anyway. 

What adverse effects could take place if I was to alias my ssh command to include the "xterm-256color" term value and login?

**Reply by page-down** [reply]
Perhaps the text will be displayed incorrectly, the keys will not work properly, etc. Causing the user to make the wrong decision about the command entered for the operation, which is then executed incorrectly. There may also be unexpected situations when other kitty-exclusive features are used.

You are on your own at this point.

You can use infocmp to compare the exact differences between terminfo.
